### PR TITLE
Sourabha G student report

### PR DESCRIPTION
**Pull request by sourabhaa**
@srevinsaju @chimosky kindly review.

### CODE REVIEW

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+*GitHub username - [sourabhaa](https://github.com/sourabhaa/)*
+
+*Mentors -  [Ibiam Chihurumnaya](https://github.com/chimosky/), [Srevin Saju](https://github.com/srevinsaju), [Jui Pradhan](https://github.com/JuiP), [Saumya Mishra](https://github.com/Saumya-Mishra9129)*
+## My work in project
+
+My work mostly revolove around adding basic game design features, porting activities to python3, test the activities in development environment and user environment(kids). Project also includes adding and fetiching new translations, fixing flake8 errors and other issues that block new release and making new release<ul><li>Game design features include altering the game flow by adding,<ol><li>A scoring system</li><li>Timer or stopwatch as a parameter to gauge win/lose(if there is no scope for score and highscore)</li><li>Levels</li><li> Displaying a UI that shows win/lose, score and highscore, replay prompt(or automatic restart of game)</li><li>Storing the game data that is used in each instance of the activity</li></ol><li>Porting activities to python3, Sugargame-v1.3</li><li>Enable to resume game from `Journal`</li><li>Test and fix logical errors in game</li></ul>
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:14]: Typo in revolve

Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:14]: Also, typo in `fetiching -> fetching`

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
...ivity by addding game-design features: <ul><li>A UI that shows the `score`, `highscore` of the player which decides the `win`/`lose`, `replay` and `exit` buttons after completion of each game.</li>Timer of 1 minute per game</li></ul><br><br> Fixed overlapping of numbers <br> enable save and restore game data from `Journal`<br> UI Enhancement: Improved the UI of main, level page and buttons <br>combined event checking for MOUSEBUTTONDOWN of game_over into play<br>Flake8 fixes| <ul><li>[Gameover UI](https://github.com/sugarlabs/jamath-activity/commit/a8779284f84095b3671a273633c2e845964be9d7)</li><li>[Event checking fix](https://github.com/sugarlabs/jamath-activity/commit/4800b9882c3f44122a51e8e7df503da0b27bf7b0)</li><li>[UI Enhancement](https://github.com/sugarlabs/jamath-activity/commit/0cc46911c57d4944ba167c743d5b0f3e826297bc)</li><li>[Flake8 fix](https://github.com/sugarlabs/jamath-activity/commit/79ee1f0a2901195526d8302e94b84e0e4e0e4472)</li><li>[Overlapping digits fix](https://github.com/sugarlabs/jamath-activity/commit/6811a0686abff9096fc909abcb7ea8745ebcc9b4)</li></ul> | [Release v4](https://github.com/sugarlabs/jamath-activity/commit/1d9bbb7c2401f5938ef08be7571536f2b3d03072)|
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:20]: Typo in `activity`

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
...each game.</li> <li>10 puzzles per game per game</li><li>Automatic restart the game after Gameover UI</li></ul> UI Enhancement: Changed UI of instruction message in each puzzle.<br> Flake8 fixes<br>PyGIWarning fix<br> Updated Sprites <br> remove unnecessary code and refactor| <ul><li>[Game over UI](https://github.com/sugarlabs/recall/commit/7ac64299554c702f9627e652cb6383cddbd844fb)</li><li>[UI Enhamcement](https://github.com/sugarlabs/recall/commit/201d4031b894c98a96e7f908ab37b67c5c25aebc)</li><li>[Save and restore game data from Journal](https://github.com/sugarlabs/recall/commit/1505dd7b51a9d061a312c3e1d133c745734a212e)</li><li>[PyGIWarning fix](https://github.com/sugarlabs/recall/commit/b9e7e3088b88f52bee2025c2237a215a325eafa9)</li><li>[Update sprites](https://github.com/sugarlabs/recall/commit/2b467d06f4a39fde0c5a8c54eb25be63c502ba2b)</li><li>[Automatic restart previous game](https://github.com/sugarlabs/recall/commit/1e7fbc26a63398b246f978d3f8cce6e3401d6848)</li><li>[Code Optimization](https://github.com/sugarlabs/recall/commit/3abf92a764b5d0bc1b5f57a1f83cf6021c576a6e)</li></ul>| [Release v8](https://github.com/sugarlabs/recall/commit/63078f026b8252f4bcba7a0ac960da0d4f5f106a)|
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:21]: Typo in `adding`

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+| Clasify-cats |Work on this activity is completed, release PR yet to be reviewed.<br>Changed flow of activity by addding game-design features: <ul><li>Added scoring system</li><li>A UI that shows the `your-score`, `high-score` of the player which decides the `win`/`lose` after completion of each game.</li><li>Altered game flow as 5 puzzles per game</li><li>Save and restore `best-time` from `Journal` in every instance of activity</li></ul>KeyError fix;fix exception handling; flake8 fixes; refactor and remove unnecessary code| <ul><li>[Gameover UI; scoring system](https://github.com/sugarlabs/classify-cats/commit/f7a66b7d605bf6904d40ddcedafc2c6270889d2a)</li><li> [save and restorhigh-score](https://github.com/sugarlabs/turtlepond/commit/1ec59cb17e4978db910cfedc771274b9ab8ad46b)</li><li> [KeyError fix](https://github.com/sugarlabs/classify-cats/commit/08935cfa66a7501eb80cbce1c158a6d44126245f)</li><li>[Fix Exception handling](https://github.com/sugarlabs/classify-cats/commit/f857a89f362206f52ad089aa5856caaa692be5c6)</li><li> [Code Optimization](https://github.com/sugarlabs/classify-cats/commit/89561168988180c18d9113892c047b5352123992)</li></ul>
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:30]: Typo in adding.

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+| Block-party |Work on this activity is complete. Release commits yet to be reviewd by the menotors| [fixed text merging with backgroun](https://github.com/sugarlabs/block-party-activity/commit/a84e7f36d26755c214e25a513d03e7c04a7833a4)|
+|Flip|(Open PR) The release of this activity is temporarily blocked by the feature 'save and restore game data from `Journal`'<br>Changed flow of activity by addding game-design features: <ul><li>A UI that shows the `your-time`, `best-time`(least time taken to solve the puzzle) of the player which decides the `win`/`lose` after completion of each game.</li><li>Stopwatch to keep track of time taken by the player to solve each puzzle</li><li>Altered game flow to 3 puzzles per level </li><li>Save and restore `best-time` from `Journal` in every instance of activity</li></ul> flake8 fixes; refactor and remove unnecessary code| <ul><li>[Gameover UI and timer](https://github.com/sugarlabs/flip/pull/13/commits/9b3d5e43ac2983c1b89ed7261a3941d0436e1941)</li><li> [save and restore best-time](https://github.com/sugarlabs/flip/pull/13/commits/216bde40bde91b008a1d29df599fffb0c08515f7)</li></ul>
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:32]: Typo in adding

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Personal note
+
+This summer was super fun and filled with learning! I enjoyed testing over 60+ activities with kids to get first-hand feedback which had a postive impact in the way they apprach problems. The fact that these activities are used by kids as a tool of learning motivates me to contribute more and more. I'm going continue to contribute and be a part of SugarLabs. Special thanks to my mentors and [James Cameron](https://github.com/quozl) for helping a lot in the initial stages to understand the codebase. A big thanks to this community which has helped to shape a lot of young lives. Lastly, thanks to Google for this wonderful initiative :)
```
Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:45]: typo in approach

Comment (srevinsaju) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md:45]: For consistency, use "Sugar Labs"

**Review by srevinsaju — CHANGES_REQUESTED** [review]
Nice write up, few typos.

**Comment by quozl** [comment]
Reviewed.  Acceptable as is.  If you have time for refinement, consider checking tense and rewriting in past tense where possible.  Also a few opportunities for punctuation and whitespace following international english conventions.

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+Improve and maintain atleast 15 Sugar Activities
```
Comment (chimosky) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md]: The project title is **Improve and Maintain 15 Activities**, no _atleast_.

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
...each game.</li> <li>10 puzzles per game per game</li><li>Automatic restart the game after Gameover UI</li></ul> UI Enhancement: Changed UI of instruction message in each puzzle.<br> Flake8 fixes<br>PyGIWarning fix<br> Updated Sprites <br> remove unnecessary code and refactor| <ul><li>[Game over UI](https://github.com/sugarlabs/recall/commit/7ac64299554c702f9627e652cb6383cddbd844fb)</li><li>[UI Enhamcement](https://github.com/sugarlabs/recall/commit/201d4031b894c98a96e7f908ab37b67c5c25aebc)</li><li>[Save and restore game data from Journal](https://github.com/sugarlabs/recall/commit/1505dd7b51a9d061a312c3e1d133c745734a212e)</li><li>[PyGIWarning fix](https://github.com/sugarlabs/recall/commit/b9e7e3088b88f52bee2025c2237a215a325eafa9)</li><li>[Update sprites](https://github.com/sugarlabs/recall/commit/2b467d06f4a39fde0c5a8c54eb25be63c502ba2b)</li><li>[Automatic restart previous game](https://github.com/sugarlabs/recall/commit/1e7fbc26a63398b246f978d3f8cce6e3401d6848)</li><li>[Code Optimization](https://github.com/sugarlabs/recall/commit/3abf92a764b5d0bc1b5f57a1f83cf6021c576a6e)</li></ul>| [Release v8](https://github.com/sugarlabs/recall/commit/63078f026b8252f4bcba7a0ac960da0d4f5f106a)|
```
Comment (chimosky) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md]: Repetition of `per game`. 

`Automatic restart the game after Gameover UI` should be `Automatically restart the game after gameover.`

**Review by chimosky — CHANGES_REQUESTED** [review]

File: archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md [hunk 9]
```diff
@@ -31,10 +32,17 @@ My work mostly revolove around adding basic game design features, porting activi
...
+
+## Personal note
+
+This summer was super fun and filled with learning! I enjoyed testing over 60+ activities with kids to get first-hand feedback which had a postive impact in the way they apprach problems. The fact that these activities are used by kids as a tool of learning motivates me to contribute more and more. I'm going continue to contribute and be a part of SugarLabs. Special thanks to my mentors and [James Cameron](https://github.com/quozl) for helping a lot in the initial stages to understand the codebase. A big thanks to this community which has helped to shape a lot of young lives. Lastly, thanks to Google for this wonderful initiative :)
```
Comment (chimosky) [inline archives/2021/student-reports/improve-and-maintain-15-activities-Sourabha-G.md]: Typo in `I'm going continue` should be `I'm going to continue`.

**Review by chimosky — CHANGES_REQUESTED** [review]

**Comment by chimosky** [comment]
Reviewed and I agree with @quozl above. Thanks for your work with the community.

**Pull request merged**